src/main/java/com/epam/ta/reportportal/commons/querygen/QueryBuilder.java

```diff
@@ -127,7 +127,8 @@ public QueryBuilder addHavingCondition(Condition condition) {
 	public QueryBuilder with(Pageable p) {
 		if (p.isPaged()) {
 			query.addLimit(p.getPageSize());
-			query.addOffset(Long.valueOf(p.getOffset()).intValue());
+			long offset = p.getOffset();
+			query.addOffset(offset < 0 ? 0 : offset > Integer.MAX_VALUE ? Integer.MAX_VALUE : (int) offset);
```

Comment:
can we move it somwere upper? in sort argument resolver for example ?

Reply:
move to method

Reply:
done